shell/server/backup.js

```diff
@@ -41,6 +41,33 @@ function recursiveRmdirIfExists(dir) {
   }
 }
 
+function walkAndWriteFiles(dir, writeStream, replaceRoot, originalRoot) {
+  originalRoot = originalRoot || dir;
+  var files = Fs.readdirSync(dir);
+  files.forEach(function (name) {
+    name = dir + "/" + name;
+
+    // Filter out file paths with a newline in them
+    if (name.indexOf("\n") === -1) {
+      return;
+    }
+
+    var fileStat = Fs.lstatSync(name);
+    if (fileStat.isDirectory()) {
+      walkAndWriteFiles(name, writeStream, replaceRoot, originalRoot);
+    } else if (fileStat.isFile()) {
+      name = replaceRoot + name.slice(originalRoot.length);
+      writeStream.write(name + "\n");
+    }
+  });
+
+  // Include empty directories
+  if (files.length === 0) {
+    dir = replaceRoot + dir.slice(originalRoot.length);
```

Comment:
Ah, makes sense. It wasn't obvious to me that this will only get called with a known entry point.